pkg/operations/cordondrainvm.go

```diff
@@ -173,24 +174,31 @@ func (o *drainOperation) deleteOrEvictPods(pods []v1.Pod) error {
 }
 
 func (o *drainOperation) evictPods(pods []v1.Pod, policyGroupVersion string) error {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Is this meant to enforce a timeout? This context object will never timeout according to a counter, its `Done()` method will only be engaged if the `cancel` return func is invoked manually.

Reply:
Yes, this change solves this problem: when timeout, we need to send a signal to the goroutines to stop and return, otherwise, it may be problematic because it will keep trying to evict the Pods forever (based on my observation in my tests). And you are right, this context object will never timeout, however, `cancel` will be invoked in `defer cancel()`, which will stop all the goroutines.